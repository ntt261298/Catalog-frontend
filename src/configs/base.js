export default {
  apiHost: 'http://localhost:5000',
};
