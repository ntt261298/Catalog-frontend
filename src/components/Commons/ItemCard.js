import React from 'react';
import { Button, Card } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

export const ItemCard = (props) => {
  const { id, title, category_id } = props.item;
  return (
    <Card key={id} className="item-card">
      <Card.Body>
        <Card.Title style={{ borderBottom: '1px solid #f7f7f7' }}>{ title }</Card.Title>
        <Link to={`/category/${category_id}/item/${id}`}>
          <Button variant="primary">
              View Detail
          </Button>
        </Link>
      </Card.Body>
    </Card>
  );
};


ItemCard.propTypes = {
  item: PropTypes.object.isRequired,
};

export default ItemCard;
