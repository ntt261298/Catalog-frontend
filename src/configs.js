import {
  configure, shallow, mount, render,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Enzyme config
configure({ adapter: new Adapter() });

// API host config
const defaultHost = 'http://localhost:5000';

export {
  shallow, mount, render, defaultHost,
};
