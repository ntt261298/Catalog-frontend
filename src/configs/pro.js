import base from 'configs/base';

export default {
  ...base,
};
