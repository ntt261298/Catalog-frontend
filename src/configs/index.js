import {
  configure, shallow, mount, render,
} from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import dev from 'configs/dev';
import pro from 'configs/pro';

// Enzyme config
configure({ adapter: new Adapter() });

let configs;

if (process.env.REACT_APP_ENV === 'dev') {
  configs = dev;
} else if (process.env.REACT_APP_ENV === 'pro') {
  console.log('pro');
  configs = pro;
} else {
  // Default configs is dev
  configs = dev;
}

export { shallow, mount, render };

export default configs;
