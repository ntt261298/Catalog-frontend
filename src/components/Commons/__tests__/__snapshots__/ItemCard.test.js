import React from 'react';
import { shallow } from 'configs';
import { ItemCard } from 'components/Commons/ItemCard';

describe('components/Commons/ItemCard', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(
      <ItemCard {...props} />,
    );
  };
  beforeEach(() => {
    props = {
      item: {
        id: 1,
        title: 'title',
        category_id: 2,
      },
    };
  });

  it('should render ItemCard', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
});
