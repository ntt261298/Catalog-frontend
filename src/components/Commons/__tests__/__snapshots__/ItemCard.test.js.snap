// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Commons/ItemCard should render ItemCard 1`] = `
<Card
  as="div"
  body={false}
  className="item-card"
  key="1"
>
  <CardBody>
    <CardTitle
      style={
        Object {
          "borderBottom": "1px solid #f7f7f7",
        }
      }
    >
      title
    </CardTitle>
    <Link
      to="/category/2/item/1"
    >
      <Button
        active={false}
        disabled={false}
        type="button"
        variant="primary"
      >
        View Detail
      </Button>
    </Link>
  </CardBody>
</Card>
`;
